ockam-network/ockam
# Crate Release 2021-12-13

- build(elixir): add script to rebuild macos native vault lib
- feat(rust): upgrade portals flow
- test(rust): add basic portal test
- feat(rust): add tcp heartbeats
- feat(rust): adjust portals delays to avoid race conditions
- feat(rust): stop tcp worker on hearbeat failure
- chore(rust): fix clippy warnings
- test(elixir): increase timeout in ordering test to make it more stable
- feat(rust): vault updates
- feat(rust): implement add key for entity
- feat(rust): update entity structure
- feat(rust): update `LocalInfo` logic
- feat(rust): add access control
- fix(rust): fix `tcp_inlet_and_outlet` example dockerfile
- ci: remove github workflows for rust and elixir
- build: improve gradle builds
- ci: add rust github workflow
- ci: add elixir github workflow
- feat(rust): introduce nested pipe behaviour test
- feat(rust): add a test for full pipe behaviour stack
- feat(rust): initial ockam channel implementation
- refactor(rust): simplify channel creation handshake
- refactor(rust): fix channel channel behavior and add tests
- fix(rust): clippy style update
- fix(rust): update channels with no_std support
- docs: remove stale ref to `KeyAttributes`
- build(rust): change all crates to `-dev` version
- ci: improve github workflow that lists gradle tasks
- feat: add suborbital integration
- feat: improve suborbital integration docker compose files
- docs: add diagram for suborbital integration
- docs: add instructions for suborbital integration
- ci: add script to get ockam publish order
- ci: move crate release scripts to ockam repo
- ci: fix publish-order script for older bash version
- ci: ensure env variables are added running script
- docs: improve ockam script documentation
- ci: align git tag release notes
- ci: exclude crates from being published
- build: rust in release mode when invoke from a parent gradle build
- build(elixir): invoke rust build only while testing
- ci: improve names and structure of commitlint workflow
- ci: improve names and structure in editorconfig workflow
- ci: replace nightly workflow with all implementation workflow
- ci: improve permissions, names and structure of gradle workflow
- ci: add test job and path triggers to elixir workflow
- ci: improve rust workflow
- docs: improve suborbital integration readme and diagram
- docs: improve guide on suborbital integration
- refactor(rust): move ockam_vault_core crate into ockam_core
- refactor(rust): change uses of `ockam_vault_core::Foo` to use `ockam_core::vault::Foo` across crates
- refactor(rust): add ockam_core/bls feature and small fixes
- ci: changelogs for release
- ci: crate release ockam version 0.42.0
- ci: crate release ockam_channel version 0.38.0
- ci: crate release ockam_command version 0.3.0
- ci: crate release ockam_core version 0.42.0
- ci: crate release ockam_entity version 0.32.0
- ci: crate release ockam_examples version 0.25.0
- ci: crate release ockam_executor version 0.11.0
- ci: crate release ockam_key_exchange_core version 0.34.0
- ci: crate release ockam_key_exchange_x3dh version 0.34.0
- ci: crate release ockam_key_exchange_xx version 0.35.0
- ci: crate release ockam_macros version 0.3.0
- ci: crate release ockam_node version 0.41.0
- ci: crate release ockam_node_no_std version 0.14.0
- ci: crate release ockam_transport_core version 0.16.0
- ci: crate release ockam_transport_tcp version 0.37.0
- ci: crate release ockam_transport_websocket version 0.31.0
- ci: crate release ockam_vault version 0.36.0
- ci: crate release ockam_vault_sync_core version 0.34.0
- ci: crate release ockam_vault_test_suite version 0.31.0
- ci: crate release signature_bbs_plus version 0.32.0
- ci: crate release signature_bls version 0.30.0
- ci: crate release signature_core version 0.32.0
- ci: crate release signature_ps version 0.30.0
- ci: remove dev tag entries



## Current Behaviour



## Proposed Changes




## Checks



- [ ] All commits in this Pull Request are [signed](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits).
- [ ] All commits in this Pull Request follow the Ockam [commit message convention](https://www.ockam.io/learn/how-to-guides/contributing/CONTRIBUTING#commit-messages).
- [ ] I accept the Ockam Community [Code of Conduct](https://www.ockam.io/learn/how-to-guides/high-performance-team/conduct/).
- [ ] I have accepted the Ockam [Contributor Licence Agreement](https://www.ockam.io/learn/how-to-guides/contributing/cla/) by adding my Git/Github details in a row at the end of the [CONTRIBUTORS.csv](https://github.com/ockam-network/contributors/blob/master/CONTRIBUTORS.csv) file in a separate pull request to the [ockam-network/contributors](https://github.com/ockam-network/contributors) repository.
